lib/auth/accountrecovery.go

```diff
@@ -308,79 +258,24 @@ func (a *Server) verifyAuthnWithRecoveryLock(ctx context.Context, startToken typ
 	case trace.IsConnectionProblem(verifyAuthnErr):
 		log.Error(trace.DebugReport(verifyAuthnErr))
 		return trace.AccessDenied(verifyRecoveryBadAuthnErrMsg)
-
 	case verifyAuthnErr == nil:
-		// Reset attempt counter.
-		if err := a.DeleteUserRecoveryAttempts(ctx, startToken.GetUser()); err != nil {
-			log.Error(trace.DebugReport(err))
-		}
-
 		return nil
 	}
 
 	log.Error(trace.DebugReport(verifyAuthnErr))
-
-	lockedUntil, maxedAttempts, err := a.recordFailedRecoveryAttempt(ctx, startToken.GetUser())
-	switch {
-	case err != nil:
-		log.Error(trace.DebugReport(err))
-		return trace.AccessDenied(verifyRecoveryBadAuthnErrMsg)
-	case !maxedAttempts:
-		return trace.AccessDenied(verifyRecoveryBadAuthnErrMsg)
-	}
-
-	// Delete all tokens related to this user, to force user to restart the recovery flow.
-	if err := a.deleteUserTokens(ctx, startToken.GetUser()); err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.AccessDenied(verifyRecoveryGenericErrMsg)
+	if err := a.recordFailedRecoveryAttempt(ctx, startToken.GetUser()); err != nil {
+		log.WithError(err).Warn("Error recording failed account recovery attempt")
 	}
 
-	// Restart the attempt counter, to not block users from trying again with another recovery code.
-	if err := a.DeleteUserRecoveryAttempts(ctx, startToken.GetUser()); err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.AccessDenied(verifyRecoveryGenericErrMsg)
-	}
-
-	// Lock the user from logging in.
-	user.SetLocked(lockedUntil, accountLockedMsg)
-	_, err = a.UpsertUser(ctx, user)
-	if err != nil {
-		log.Error(trace.DebugReport(err))
-		return trace.AccessDenied(verifyRecoveryBadAuthnErrMsg)
-	}
-
-	return trace.AccessDenied(MaxFailedAttemptsFromVerifyRecoveryErrMsg)
+	return trace.AccessDenied(verifyRecoveryBadAuthnErrMsg)
 }
 
-// recordFailedRecoveryAttempt creates and inserts a recovery attempt and if user has reached max failed attempts,
-// returns the locked until time. The boolean determines if user reached maxed failed attempts (true) or not (false).
-func (a *Server) recordFailedRecoveryAttempt(ctx context.Context, username string) (time.Time, bool, error) {
-	maxedAttempts := true
-
+// recordFailedRecoveryAttempt creates and inserts a recovery attempt and if user has reached max failed attempts.
+func (a *Server) recordFailedRecoveryAttempt(ctx context.Context, username string) error {
```

Comment:
Do we even need this anymore? What is the point of recording a failed audit attempt if we no longer need to know how many failed attempts there have been?

The failed attempt is already written to the audit log, so I suspect this can be removed.

Reply:
I removed the attempt CRUD API fully now, thank you for the help here